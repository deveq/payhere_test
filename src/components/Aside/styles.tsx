import styled from 'styled-components';
import {flexCenter} from 'styles/helper';

export const Aside = styled.aside`
    width: 200px;
    background: ${({theme}) => theme.bg_page1};
    padding: 30px;
    box-sizing: border-box;
    flex-shrink: 0;
`;

export const List = styled.ul`
    display: grid;
    gap: 5px;
`;

export const Title = styled.h2`
    font-size: 24px;
    font-weight: bold;
    border-bottom: ${({theme}) => `1px solid ${theme.bg_element4}`};
    color: ${({theme}) => theme.text1};
    padding-bottom: 10px;
    margin-bottom: 20px;
    text-indent: 10px;
`;

export const ListItem = styled.li<{selected: boolean}>`
    background: ${({selected,theme}) => selected ? theme.bg_element1 : 'transparent'};
    padding: 15px 20px;
    border-radius: 10px;
    font-weight: bold;
    color: ${({selected, theme}) => selected ? theme.text2 : theme.text3};
    &:hover,
    &:active {
        background: ${({theme}) => theme.bg_element1};
    };
    cursor: pointer;
`;

export const Subtitle = styled.div`

`;