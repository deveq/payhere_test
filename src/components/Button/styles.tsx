import styled from "styled-components";

type ButtonSize = 'sm' | 'md' | 'lg';
export interface ButtonProps {
    size?: ButtonSize;
    backgroundColor?: string;
    borderColor?: string;
    color?: string;
    showShadow?: boolean;
}

const handleSize = (size?: ButtonSize) => {
    if (size === 'sm') return '0.5rem 1rem';
    if (size === 'md') return '0.75rem 1.5rem';
    if (size === 'lg') return '1rem 2rem';
    return '0.75rem 1.5rem';
}

export const Button = styled.button<ButtonProps>`
    background: ${({backgroundColor, theme}) => 
        backgroundColor || theme.bg_element2
    };
    border-radius: 5px;
    color: ${({color, theme}) => 
        color || theme.text1
    };
    padding: ${({size}) => handleSize(size)};
    font-weight: bold;
    box-shadow: ${({theme, showShadow}) => showShadow ? theme.boxShadow : 'none'};
    border: ${({showShadow, theme}) => showShadow ? 'none' : `1px solid ${theme.border2}` };
    &:hover,
    &:active {
        background: ${({theme}) => theme.bg_element3};
    }
    cursor: pointer;
`;

Button.defaultProps = {
    size: 'md',
}

export const Primary = styled(Button)`
    background: ${({theme}) => theme.primary};
    color: white;
`;

export const Destructive = styled(Button)`
    background: ${({theme}) => theme.destructive};
    color: white;
    border: none;
    &:hover,
    &:active {
        background: ${({theme}) => theme.destructive};
    }
`;