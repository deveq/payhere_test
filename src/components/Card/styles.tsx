import styled from 'styled-components';

export const CardContainer = styled.div`
    background: ${({theme}) => theme.bg_element1};
    border-radius: 10px;
    padding: 5px 10px;
`;