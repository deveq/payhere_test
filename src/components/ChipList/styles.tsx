import styled from 'styled-components';

interface ChipProps {
    selected?: boolean;
}

export const Chip = styled.div<ChipProps>`
    padding: 5px 10px;
    max-width: 150px;
    min-width: 40px;
    text-align: center;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    border-radius: 20px;
    border: 1px solid ${({theme}) => theme.border2};
    background: ${({selected, theme}) => selected ? theme.primary : 'white'};
    color: ${({selected}) => selected ? 'white' : '#333'};
    cursor: pointer;
    &:active {
        opacity: 0.9;
    }
`;

export const ChipList = styled.div`
    display: flex;
    /* flex-direction: column;
    align-items: flex-start; */
    flex-wrap: wrap;
    gap: 8px;
    margin-bottom: 10px;
`;

export const Message = styled.p`
    color: ${({theme}) => theme.text1};
`;