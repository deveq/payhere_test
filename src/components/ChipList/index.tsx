import * as Styled from './styles';

interface ChipListProps {
	data: string[];
	onClick?: (text: string) => void;
    selectedItem?: string;
}

const ChipList = ({ data, onClick, selectedItem }: ChipListProps) => {
	return (
		<Styled.ChipList>
			{data.map((item) => (
				<Styled.Chip
					key={item}
					selected={selectedItem === item}
					onClick={() => {
						if (onClick) onClick(item);
					}}
				>
					{item}
				</Styled.Chip>
			))}
		</Styled.ChipList>
	);
};

export default ChipList;
