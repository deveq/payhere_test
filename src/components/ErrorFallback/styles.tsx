import styled from 'styled-components';
import {flexCenter} from 'styles/helper';
import {FaRegSadTear} from 'react-icons/fa';

export const Container = styled.div`
    height: 100vh;
    ${flexCenter};
    flex-direction: column;
`;

export const Paragraph = styled.p`
    font-size: 30px;
`;

export const ErrorMessage = styled.pre`
    font-size: 20px;
`;

export const SadIcon = styled(FaRegSadTear)`
    width: 100px;
    height: 100px;
    color: gray;
`;