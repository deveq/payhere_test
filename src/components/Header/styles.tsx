import React from 'react';
import styled from 'styled-components';
import { ReactComponent } from 'assets/logo.svg';
import {flexCenter} from 'styles/helper';


export const Container = styled.header`
	display: flex;
	padding: 1rem 2rem;
	background: ${(props) => props.theme.bg_page1};
	box-shadow: ${(props) => props.theme.boxShadow};
`;

export const Logo = styled(ReactComponent)`
	width: 50px;
	height: 50px;
	${flexCenter};
`;

export const Headline = styled.h1`
`;

export const Nav = styled.nav`
	flex: 1;
	${flexCenter};
`;

export const LoginContainer = styled.div`

`;