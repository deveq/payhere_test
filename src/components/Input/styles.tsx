import { Children } from 'react';
import styled from 'styled-components';
import { FiSearch } from 'react-icons/fi';
import { flexCenter } from 'styles/helper';

interface HistoryProps {
	show?: boolean;
}

export const Container = styled.form`
	${flexCenter};
	width: 250px;
	border-radius: 10px;
	margin-left: 20px;
	padding: 10px;
	border: 1px solid ${({ theme }) => theme.border3};
	background: ${({ theme }) => theme.bg_element1};
	position: relative;
`;

export const Input = styled.input`
	border: none;
	outline: none;
	flex: 1;
	font-size: 16px;
	color: ${({ theme }) => theme.text1};
	background: ${({ theme }) => theme.bg_element1};
`;

export const Icon = styled(FiSearch).attrs({
	'data-testid': 'icon'
})`
	width: 20px;
	height: 20px;
	color: ${({ theme }) => theme.text1};
	cursor: pointer;
	&:hover,
	&:active {
		opacity: 0.6;
	}
`;

export const History = styled.ul<HistoryProps>`
	display: ${({ show }) => (show ? 'flex' : 'none')};
	position: absolute;
	top: 45px;
	background: ${({ theme }) => theme.bg_element1};
	width: 100%;
	height: ${({ children }) => `${Children.count(children) * 40}`}px;
	border-radius: 10px;
	z-index: 10;
	box-shadow: ${({ theme }) => theme.boxShadow};
	padding: 10px 20px;
	box-sizing: border-box;
	flex-direction: column;
	justify-content: space-around;
`;

export const HistoryItem = styled.li`
	color: ${({ theme }) => theme.text1};
	font-size: 18px;
	display: flex;
	justify-content: space-between;
	max-width: 90%;
	white-space: nowrap;
	text-overflow: ellipsis;
	overflow: hidden;
	&:hover {
		opacity: 0.6;
	}
	cursor: pointer;
`;
