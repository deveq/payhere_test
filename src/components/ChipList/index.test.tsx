import {screen, fireEvent} from '@testing-library/react';
import {renderWithWrapper} from 'tests';
import ChipList from './index';
import {lightTheme} from 'styles/theme';

describe('<ChipList />', () => {
    const data = ['a','b','c'];
    const emptyData: string[] = [];

    it('정상적으로 렌더링 되는지 확인', () => {
        const {container} = renderWithWrapper(
            <ChipList 
                data={data}
            />
        )
        const chip = screen.getByText(data[0]);
        const chipList = chip.parentElement;
        expect(chipList).toBeInTheDocument();
        expect(chip).toBeInTheDocument();
        expect(chipList?.childElementCount).toBe(data.length);
        expect(container).toMatchSnapshot();
    });

    it('ChipList의 스타일이 정상적으로 렌더링 되었는지 확인한다.', () => {
        renderWithWrapper(
            <>
                <ChipList 
                    data={data}
                />
            </>
        )
        const chip = screen.getByText(data[0]);
        const chipList = chip.parentElement;
        expect(chipList).toHaveStyle({
            display: 'flex',
            'flex-wrap': 'wrap',
            'gap': '8px',
            'margin-bottom': '10px',
        });
    });

    it('Chip의 스타일이 정상적으로 렌더링 되었는지 확인한다', () => {
        renderWithWrapper(
            <>
                <ChipList 
                    data={data}
                    selectedItem={data[1]}
                />
            </>
        )
        const chip = screen.getByText(data[0]);
        const selectedChip = screen.getByText(data[1]);
        expect(chip).toHaveStyle({
            padding: '5px 10px',
            'max-width': '150px',
            'text-align': 'center',
            overflow: 'hidden',
            'text-overflow': 'ellipsis',
            'white-space': 'nowrap',
            'border-radius': '20px',
            border: `1px solid ${lightTheme.border2}`,
            cursor: 'pointer',

            background: 'white',
            color: '#333',
        });

        expect(selectedChip).toHaveStyle({
            background: lightTheme.primary,
            color: 'white',
        })
    });

    it('Message가 정상적으로 렌더링되는지 확인한다', () => {
        renderWithWrapper(<ChipList 
            data={emptyData}
        />);
        
        const message = screen.getByText('관심 목록을 추가해주세요~');
        expect(message).toBeInTheDocument();
        expect(message).toHaveStyle({
            color: lightTheme.text1
        });
    })

    it('click 이벤트가 정상동작하는지 확인한다', () => {
        const fn = jest.fn();
        renderWithWrapper(<ChipList data={data} onClick={fn} />);
        const chip = screen.getByText(data[0]);
        expect(fn).toHaveBeenCalledTimes(0);
        fireEvent.click(chip);
        expect(fn).toHaveBeenCalledTimes(1);
    })
})