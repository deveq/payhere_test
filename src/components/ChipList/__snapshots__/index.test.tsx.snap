// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChipList /> 정상적으로 렌더링 되는지 확인 1`] = `
<div>
  <div
    class="sc-gswNZR mmBBh"
  >
    <div
      class="sc-bcXHqe jhqMJh"
    >
      a
    </div>
    <div
      class="sc-bcXHqe jhqMJh"
    >
      b
    </div>
    <div
      class="sc-bcXHqe jhqMJh"
    >
      c
    </div>
  </div>
</div>
`;
